### PR TITLE
Rework Testing Location

### LINKED ISSUE #870: Reduce CircleCI build time by downloading/caching frameworks.
I'm not very familiar with the CircleCI / Rake setup, but I think we can reduce our CI build times by downloading RxSwift instead of building it. I can't find any `--no-use-binaries` flags for carthage, so I don't know why it isn't downloading the framework in the first place. Perhaps someone with more experience with the CI setup will have more luck.

### PR DESCRIPTION
**Pull request by scottrhoyt**
This PR is meant to aid with #885. Also Closes #870. This is what is does:

* moves `Tests` to root dir
* moves test targets from Demo.xcworkspace to Moya.xcodeproj
* switches CircleCI to run tests from the main project
* adds a CircleCI task to build the Demo project
* greatly speeds up build time by caching carthage builds (~ 50%)
* fixes an error where `rake test:carthage` wasn't actually testing the current commit

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -92,7 +92,10 @@ end
 
 desc 'Bootstrap Carthage.'
 task :bootstrap do
-  safe_sh "carthage bootstrap"
```
Comment (AndrewSB) [inline Rakefile]: I think this can be done in one line, `carthage bootstrap --platform ios tvOS macOS`

**Comment by AndrewSB** [comment]
What I understand from this is 
1. You moved the tests into Moya.xcodeproj
2. You actually got the tests to run & updated Rake things, and cocoapods things, and implemented caching

If we could cherrypick all the 2 commits from this branch onto `redo-spm-layout`, and then merge the PR into #885, then we'll have this PR done, and 2/3 of the TODOs on #855 done as well

**Comment by scottrhoyt** [comment]
It's a little bit more than that. Let me take a stab at combining them. I know how to make the changes again. I'll push the commits right to your branch.

**Comment by AndrewSB** [comment]
awesome, I'll take a look 👍

**Comment by BasThomas** [comment]
I've deleted the branch; please restore if needed!

**Comment by scottrhoyt** [comment]
All good. Thanks. Uneeded now.

**Pull request closed without merging**